mitiq/cdr/clifford_training_data.py

```diff
@@ -0,0 +1,488 @@
+    )
+    r_z_gates = operations[zgatesmask]
+    r_z_positions = positions[zgatesmask]
+    r_z_qubits = qubits[zgatesmask]
+    angles = _get_arguments(r_z_gates)
+    mask_non_cliff = ~_is_clifford_angle(angles)
+    rz_non_cliff = angles[mask_non_cliff]
+    pos_non_cliff = r_z_positions[mask_non_cliff]
+    qubits_non_cliff = r_z_qubits[mask_non_cliff]
+    for n in range(num_training_circuits):
+        random_state = random_states[n]
+        random_state = np.random.RandomState(random_state)
+        #  array:
+        if additional_options:
+            (projected_circuit) = _map_to_near_clifford(
+                operations.copy(),
+                rz_non_cliff.copy(),
+                pos_non_cliff.copy(),
+                qubits_non_cliff.copy(),
+                fraction_non_clifford,
+                random_state,
+                method_select,
+                method_replace,
+                additional_options=additional_options.get(
+                    "additional_options"
+                ),
+            )
+        else:
+            (projected_circuit) = _map_to_near_clifford(
+                operations.copy(),
+                rz_non_cliff.copy(),
+                pos_non_cliff.copy(),
+                qubits_non_cliff.copy(),
+                fraction_non_clifford,
+                random_state,
+                method_select,
+                method_replace,
+            )
+        circuits_list.append(projected_circuit)
+    return circuits_list
+
+
+def _map_to_near_clifford(
+    operations: np.ndarray,
+    rz_non_cliff: np.ndarray,
+    pos_non_cliff: np.ndarray,
+    qubits_non_cliff: np.ndarray,
+    fraction_non_clifford: float,
+    random_state: np.random.RandomState,
+    method_select: str = "uniform",
+    method_replace: str = "closest",
+    **additional_options: dict,
+) -> Tuple[Circuit, List[float], List[float]]:
+    """Function to take the information in some circuit of interest and
+        return a near-Clifford circuit as constructed according to
+        some user defined methods.
+    Args:
+        circuit: circuit of interest.
+        fraction_non_clifford: the fraction of non-Clifford gates to replace
+                               in the circuit of interest.
+        method_select: string defining the method used to select the
+                       non-Clifford gates to replace: 'uniform' or
+                       'gaussian'.
+        method_replace: string defining method used to replace selected
+                        non-Clifford gates: 'closest', 'uniform' or
+                        'gaussian'.
+        random_state: Seed for sampling.
+        sigma_select: width of probability distribution used in selection
+                      of non-Clifford gates to replace, only has effect if
+                      method_select = 'gaussian'
+        sigma_replace: width of probability distribution used in replacement
```

Comment:
The docstring seems different from the args in the signature, they need to be updated?

Reply:
Yes, thanks for spotting.